tap/extensions/http/main_test.go

```diff
@@ -0,0 +1,312 @@
+package http
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/up9inc/mizu/tap/api"
+)
+
+const (
+	binDir          = "bin"
+	patternBin      = "*_req.bin"
+	patternDissect  = "*.json"
+	msgDissecting   = "Dissecting:"
+	msgAnalyzing    = "Analyzing:"
+	msgRepresenting = "Representing:"
+	respSuffix      = "_res.bin"
+	expectDir       = "expect"
+	dissectDir      = "dissect"
+	analyzeDir      = "analyze"
+	representDir    = "represent"
+	testUpdate      = "TEST_UPDATE"
+)
+
+func TestRegister(t *testing.T) {
+	dissector := NewDissector()
+	extension := &api.Extension{}
+	dissector.Register(extension)
+	assert.Equal(t, "http", extension.Protocol.Name)
+	assert.NotNil(t, extension.MatcherMap)
+}
+
+func TestMacros(t *testing.T) {
+	expectedMacros := map[string]string{
+		"http":  `proto.name == "http" and proto.version.startsWith("1")`,
+		"http2": `proto.name == "http" and proto.version == "2.0"`,
+		"grpc":  `proto.name == "http" and proto.version == "2.0" and proto.macro == "grpc"`,
+	}
+	dissector := NewDissector()
+	macros := dissector.Macros()
+	assert.Equal(t, expectedMacros, macros)
+}
+
+func TestPing(t *testing.T) {
+	dissector := NewDissector()
+	dissector.Ping()
+}
+
+func TestDissect(t *testing.T) {
+	var testUpdateEnabled bool
+	_, present := os.LookupEnv(testUpdate)
+	if present {
+		testUpdateEnabled = true
+	}
```

Comment:
```suggestion
	_, testUpdateEnabled := os.LookupEnv(testUpdate)
```

Reply:
I see it in the code and I think it is doing the same thing..

Reply:
@gadotroee done with https://github.com/up9inc/mizu/commit/4b6b07bf26df330d014316eaad35b7b83b22067a